src/main/java/assets/game.js

```diff
@@ -281,6 +286,62 @@ function initGame() {
         }
     });
 
+    document.getElementById("moveNorthButton").addEventListener("click", function(e) {
+        if(numMovesLeft !== 0) {
+            numMovesLeft--;
+            sendXhr("POST", "/move", {game: game, direction: 0}, function(data) {
+                game = data;
+                redrawGrid();
+                sonarCount--;
+                isSonar = false;
+            });
+        } else {
+            printMsgCustom('You can only use 2 moves', "red");
+        }
+    });
+
```

Comment:
Great, there is some code repetition, however this is harder to avoid in this case.